app/components/course_details/view.rb

```diff
@@ -91,19 +112,25 @@ def course_type
     end
 
     def course_start_date
-      return @not_provided_copy if data_model.course_start_date.blank?
-
-      date_for_summary_view(data_model.course_start_date)
+      date_for_summary_view(data_model.course_start_date) if data_model.course_start_date.present?
     end
 
     def course_end_date
-      return @not_provided_copy if data_model.course_end_date.blank?
-
-      date_for_summary_view(data_model.course_end_date)
+      date_for_summary_view(data_model.course_end_date) if data_model.course_end_date.present?
     end
 
     def course
       @course ||= Course.find_by(code: data_model.course_code)
     end
+
+    def mappable_field(field_value, field_label)
+      MappableFieldRow.new(
+        field_value: field_value,
+        field_label: field_label,
+        text: t("components.confirmation.missing"),
+        action_url: edit_trainee_course_details_path(trainee),
+        has_errors: has_errors,
```

Comment:
I added a `apply_draft` option to `MappableFieldRow`. If it's true, it will render the correct hint text for an apply draft eg `Review this answer` instead of `Enter an answer` (for all non apply_drafts). Since this component is used for the apply draft sections i think we need to pass in the flag here `apply_draft: trainee.apply_application?`